exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/QueryResources.java

```diff
@@ -124,7 +126,7 @@ public QueryPage(WorkManager work, HttpServletRequest request) {
       onlyImpersonationEnabled = WebServer.isOnlyImpersonationEnabled(config);
       autoLimitEnabled = config.getBoolean(ExecConstants.HTTP_WEB_CLIENT_RESULTSET_AUTOLIMIT_CHECKED);
       defaultRowsAutoLimited = config.getInt(ExecConstants.HTTP_WEB_CLIENT_RESULTSET_AUTOLIMIT_ROWS);
-      csrfToken = Utilities.getCsrfTokenFromHttpRequest(request);
+      csrfToken = Optional.ofNullable(WebUtils.getCsrfTokenFromHttpRequest(request)).orElse("");
```

Comment:
I am still unclear why token should be defaulted to empty String...
Why we can't just return it as null?

Reply:
We need to inject something in web pages, even if it is null.

Reply:
Well, then I guess it's better to return to your previous solution where returned `WebUtils.getCsrfTokenFromHttpRequest` empty string when null...

Reply:
Done.